drivers/gles3/rasterizer_storage_gles3.cpp

```diff
@@ -5895,7 +5895,10 @@ void RasterizerStorageGLES3::update_particles() {
 							tex = resources.white_tex;
 						} break;
 					}
+				} else if (t->proxy && t->proxy->tex_id) {
 
+					target = t->proxy->target;
+					tex = t->proxy->tex_id;
 				} else {
 
 					target = t->target;
```

Comment:
You could have used `t->get_ptr()->target` here instead, not adding a new else path. The `get_ptr()` checks if a proxy exists and returns it in that case, otherwise returns `this`. I think that is used in some other places as well, so might be worth using that instead.

Reply:
ah, that explains why I could only find the fix in the sky shader.
I'm about to go to sleep, if no-one has fixed this by tomorrow I'll change both the sky shader and this to use get_ptr()